test/static.test.js

```diff
@@ -566,123 +470,103 @@ t.test('register /static and /static2', (t) => {
     rep.sendFile('bar.html')
   })
 
-  t.teardown(fastify.close.bind(fastify))
+  t.after(() => fastify.close())
 
-  fastify.listen({ port: 0 }, (err) => {
-    t.error(err)
+  await fastify.listen({ port: 0 })
 
-    fastify.server.unref()
+  fastify.server.unref()
 
-    t.test('/static/index.html', (t) => {
-      t.plan(4 + GENERIC_RESPONSE_CHECK_COUNT)
-      simple.concat({
-        method: 'GET',
-        url: 'http://localhost:' + fastify.server.address().port + '/static/index.html'
-      }, (err, response, body) => {
-        t.error(err)
-        t.equal(response.statusCode, 200)
-        t.not(body.toString(), index2Content)
-        t.equal(body.toString(), indexContent)
-        genericResponseChecks(t, response)
-      })
-    })
+  await t.test('/static/index.html', async (t) => {
+    t.plan(4 + GENERIC_RESPONSE_CHECK_COUNT)
 
-    t.test('/static/bar.html', (t) => {
-      t.plan(3 + GENERIC_RESPONSE_CHECK_COUNT)
-      simple.concat({
-        method: 'GET',
-        url: 'http://localhost:' + fastify.server.address().port + '/static/bar.html'
-      }, (err, response, body) => {
-        t.error(err)
-        t.equal(response.statusCode, 200)
-        t.equal(body.toString(), barContent)
-        genericResponseChecks(t, response)
-      })
-    })
+    const response = await fetch('http://localhost:' + fastify.server.address().port + '/static/index.html')
+    t.assert.ok(response.ok)
+    t.assert.deepStrictEqual(response.status, 200)
+    const responseText = await response.text()
+    t.assert.deepStrictEqual(responseText, indexContent)
+    t.assert.notEqual(responseText, index2Content)
```

Comment:
```suggestion
    t.assert. notStrictEqual(responseText, index2Content)
```

Reply:
Could you remove the space? So I can accept the proposed change